### PR TITLE
Added Mget and Mset commands.

### PR DESCRIPTION
**Pull request by SanHalacogluImproving**

### CODE REVIEW

**Review by Elen-Ghulam — APPROVED** [review]

File: java/client/src/main/java/glide/api/BaseClient.java [hunk 1]
```diff
@@ -145,6 +149,10 @@ protected String handleStringOrNullResponse(Response response) throws RedisExcep
         return handleRedisResponse(String.class, true, response);
     }
 
+    protected Object[] handleArrayReponse(Response response) {
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/BaseClient.java:152]:
```suggestion
    protected Object[] handleArrayReponse(Response response) throws RedisException {
```

File: java/client/src/main/java/glide/api/BaseClient.java [hunk 2]
```diff
@@ -173,4 +181,26 @@ public CompletableFuture<String> set(
...
+        return commandManager
+                .submitNewCommand(MGet, keys, this::handleArrayReponse)
+                .thenApply(
+                        objectsArray ->
+                                Arrays.stream(objectsArray).map(object -> (String) object).toArray(String[]::new));
+    }
+
+    @Override
+    public CompletableFuture<String> mset(@NonNull Map<String, String> keyValueMap) {
+        String[] args = new String[keyValueMap.size() * 2];
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/BaseClient.java:196]: You should be able to convert the entryset to an array using `stream().flatmap()`
But I don't know the exact code.  You will need to put the key, value into another stream and then toArray at the end.

File: java/client/src/main/java/glide/api/commands/StringCommands.java [hunk 3]
```diff
@@ -48,4 +49,24 @@ public interface StringCommands {
      *     is set, return the old value as a <code>String</code>.
      */
     CompletableFuture<String> set(String key, String value, SetOptions options);
+
+    /**
+     * Retrieve the values of multiple <code>keys</code>.
+     *
+     * @see <a href="https://redis.io/commands/mget/">redis.io</a> for details.
+     * @param keys A list of keys to retrieve values for.
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/StringCommands.java:57]:
```suggestion
     * @param keys A list of <code>keys</code> to retrieve values for.
```

File: java/client/src/main/java/glide/api/commands/StringCommands.java [hunk 4]
```diff
@@ -48,4 +49,24 @@ public interface StringCommands {
...
+     *     </code>.
+     */
+    CompletableFuture<String[]> mget(String[] keys);
+
+    /**
+     * Set multiple keys to multiple values in a single operation.
+     *
+     * @see <a href="https://redis.io/commands/mset/">redis.io</a> for details.
+     * @param keyValueMap A key-value map consisting of keys and their respective values to set.
+     * @return Always <code>"Ok"</code>.
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/commands/StringCommands.java:69]: I thought we were going to consistently include "Response from Redis containing..." in the response?

File: java/client/src/test/java/glide/api/RedisClientTest.java [hunk 5]
```diff
@@ -201,4 +205,52 @@ public void set_with_SetOptions_OnlyIfDoesNotExist_returns_success() {
         assertNotNull(response);
         assertEquals(value, response.get());
     }
+
+    @SneakyThrows
+    @Test
+    public void mget_returns_success() {
+        // setup
+        String[] keys = {"Key1", "Key2"};
+        String[] values = {"Value1", "Value2"};
```
Comment (acarbonetto) [inline java/client/src/test/java/glide/api/RedisClientTest.java:214]: add a null value in here?

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 6]
```diff
@@ -247,4 +251,21 @@ public void set_missing_value_and_returnOldValue_is_null(BaseClient client) {
...
+        String key2 = RandomStringUtils.randomAlphabetic(10);
+        String key3 = RandomStringUtils.randomAlphabetic(10);
+        String nonExisting = RandomStringUtils.randomAlphabetic(10);
+        String value = RandomStringUtils.randomAlphabetic(10);
+        Map<String, String> keyValueMap = Map.of(key1, value, key2, value, key3, value);
+
+        assertEquals(OK, client.mset(keyValueMap).get(10, SECONDS));
+        assertArrayEquals(
+                new String[] {value, value, null, value},
+                client.mget(new String[] {key1, key2, nonExisting, key3}).get(10, SECONDS));
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:269]:
```suggestion
                client.mget(new String[] {key1, key2, nonExisting, key3}).get());
```
`@Timeout(10)` is specified in the file which will timeout any test that last longer than 10 seconds.

**Review by jonathanl-bq — APPROVED** [review]

File: java/client/src/main/java/glide/api/commands/StringCommands.java (hunk 3 above)
Comment (SanHalacogluImproving) [reply java/client/src/main/java/glide/api/commands/StringCommands.java:57]: Followed their documentation.

**Review by Yury-Fridlyand — APPROVED** [review]

File: java/client/src/main/java/glide/api/BaseClient.java [hunk 7]
```diff
@@ -173,4 +182,22 @@ public CompletableFuture<String> set(
...
+                .thenApply(
+                        objectsArray ->
+                                Arrays.stream(objectsArray).map(object -> (String) object).toArray(String[]::new));
+    }
+
+    @Override
+    public CompletableFuture<String> mset(@NonNull Map<String, String> keyValueMap) {
+        String[] args =
+                keyValueMap.entrySet().stream()
+                        .flatMap(entry -> Stream.of(entry.getKey(), entry.getValue()))
```
Comment (acarbonetto) [inline java/client/src/main/java/glide/api/BaseClient.java:199]: ❤️

**Review by acarbonetto — APPROVED** [review]

File: java/client/src/test/java/glide/api/RedisClientTest.java [hunk 8]
```diff
@@ -234,15 +236,15 @@ public void mget_returns_success() {
     @Test
     public void mset_returns_success() {
         // setup
-        Map<String, String> keyValueMap = Map.of("key1", "value1");
-
-        String[] flattenedKeyValueMap = {"key1", "value1"};
+        Map<String, String> keyValueMap = Map.of("key1", "value1", "key2", "value2");
```
Comment (Yury-Fridlyand) [inline java/client/src/test/java/glide/api/RedisClientTest.java:239]: You can use `LinkedHashMap` to ensure the order and avoid extra depedencies.

**Review by jonathanl-bq — APPROVED** [review]
Get rid of the hamcrest dependency and I think it looks good.

File: java/client/src/main/java/glide/api/RedisClient.java [hunk 9]
```diff
@@ -41,7 +41,7 @@ public CompletableFuture<Object> customCommand(@NonNull String[] args) {
 
     @Override
     public CompletableFuture<Object[]> exec(Transaction transaction) {
-        return commandManager.submitNewCommand(transaction, this::handleArrayResponse);
+        return commandManager.submitNewCommand(transaction, this::handleArrayOrNullResponse);
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/RedisClient.java:44]: Why/When could be null?

Comment (acarbonetto) [reply java/client/src/main/java/glide/api/RedisClient.java:44]: https://github.com/Bit-Quill/glide-for-redis/blob/b324a711cf7b7089f46dbf4b175b959ac55b04bd/java/client/src/main/java/glide/api/commands/GenericClusterCommands.java#L71C1-L73C1

Comment (acarbonetto) [reply java/client/src/main/java/glide/api/RedisClient.java:44]: > If the transaction failed due to a <code>WATCH</code> command, <code>exec</code> will return <code>null</code>.

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**